app/models/pipeline_run.rb

```diff
@@ -1445,10 +1445,15 @@ def get_summary_contig_counts(min_contig_size)
 
   def get_summary_contig_counts_v2(min_contig_size)
     summary_dict = {} # key: count_type:taxid , value: contigs, contig_reads
-    contig_lineages(min_contig_size).each do |c|
-      lineage = JSON.parse(c.lineage_json)
-      lineage.each do |count_type, taxid_arr|
-        taxids = taxid_arr[0..1]
+    contig_taxids = contigs.select("read_count, species_taxid_nt, species_taxid_nr, genus_taxid_nt, genus_taxid_nr")
```

Comment:
You might want to try pluck instead of select.
Creating the activerecord model object takes time and is not necessary here.

Reply:
Done.